blocks/aem-asset-selector/aem-asset-selector-util.js

```diff
@@ -55,62 +155,215 @@ export function init(cfg, callback) {
-  const selectedAsset = selection[0];
+/**
+ * Retrieves the rendition that should be copied into the clipboard for
+ * the given asset.
+ * @param {Asset} asset Asset whose copy rendition should be retrieved.
+ * @returns {Rendition} Rendition whose content should be copied to
+ *  the clipboard. Will return a falsy value if no suitable rendition
+ *  could be found.
+ */
+function getCopyRendition(asset) {
   let maxRendition = null;
-  // eslint-disable-next-line no-underscore-dangle
-  selectedAsset._links['http://ns.adobe.com/adobecloud/rel/rendition'].forEach((rendition) => {
-    if ((!maxRendition || maxRendition.width < rendition.width)) {
-      maxRendition = rendition;
-    }
-  });
+  const renditions = getRel(asset, REL_RENDITIONS);
+  if (!renditions) {
+    return maxRendition;
+  }
+  renditions
+    // TODO: the clipboard API only supports using PNGs as the blob, so
+    // only using PNG renditions. Will be fixed to allow altnernate
+    // formats in a follow-up ticket
+    .filter((rendition) => rendition.type === 'image/png')
+    .forEach((rendition) => {
+      if ((!maxRendition || maxRendition.width < rendition.width)) {
+        maxRendition = rendition;
+      }
+    });
+  return maxRendition;
+}
 
-  const assetPublicUrl = await getAssetPublicUrl(maxRendition.href.substring(0, maxRendition.href.indexOf('?')));
-  console.log('Asset public URL:', assetPublicUrl);
+/**
+ * Uses the navigator global object to write a clipboard item to the clipboard.
+ * The clipboard item's content will be an <img /> tag with the given URL as
+ * its src property.
+ * @param {string} assetPublicUrl URL to use as the image's src.
+ * @returns {Promise} Resolves when the item has been written to the clipboard.
+ */
+async function copyToClipboardWithHtml(assetPublicUrl) {
   const assetMarkup = `<img src="${assetPublicUrl}"  />`;
 
   const data = [
     // eslint-disable-next-line no-undef
     new ClipboardItem({ 'text/html': new Blob([assetMarkup], { type: 'text/html' }) }),
   ];
   // Write the new clipboard contents
-  await navigator.clipboard.write(data);
-  // onClose();
+  return navigator.clipboard.write(data);
 }
 
-// eslint-disable-next-line no-unused-vars
-function handleNavigateToAsset(asset) {
-  // onClose();
+/**
+ * Uses the navigator global object to write a clipboard item to the clipboard.
+ * The clipboard item's content will be a Blob containing the image binary from
+ * the given URL, which the method will retrieve.
+ * @param {string} assetPublicUrl URL of the image to retrieve.
+ * @param {string} mimeType Content type of the image being retrieved.
+ * @returns {Promise} Resolves when the item has been written to the clipboard.
+ */
+async function copyToClipboardWithBinary(assetPublicUrl, mimeType) {
+  const binary = await fetch(assetPublicUrl);
+
+  if (!binary.ok) {
+    throw new Error(`Unexpected status code ${binary.status} retrieving asset binary`);
+  }
+
+  const blob = await binary.blob();
```

Comment:
Two things:
- I would add a try/catch statement here to handle errors,
- I would verify the blob is not null/undefined ("empty").